### ISSUE TITLE
 PROJ_LENGTH_UNIT for inverse transforms reverts from cm back to inch

### ISSUE THREAD

**Issue by GregNeumann** [GenericMappingTools/gmt#7220]
GMT mapproject forward transform, e.g. polar stereographic, outputs in units of cm (times specified scale), as specified in gmt.conf, however the inverse transform "mapproject -I" appears to input in units of inch.
# Projection Parameters
#
PROJ_AUX_LATITUDE              = none
PROJ_DATUM                     = 
PROJ_ELLIPSOID                 = 2439400
PROJ_GEODESIC                  = Vincenty
PROJ_LENGTH_UNIT               = cm
PROJ_MEAN_RADIUS               = authalic
PROJ_SCALE_FACTOR              = 1

echo "90 89" | mapproject -Js0/90/1:100 -C -R0/360/55/90
42576.6425776	-2.60706745255e-12

echo "42576.6425776 -2.60706745255e-12"|mapproject -Js0/90/1:100 -C -R0/360/55/90 -I
90	89.6062907675

echo 90 89.6062907675 |mapproject -Js0/90/1:100 -C -R0/360/55/90
16762.4577062	-1.02640450879e-12

**Expected outcome**
Inverse should match original

Appears to have mangled meters and inches, since the x distance on Mercury is correct in meters, but 
the result is scaled down by 2.54
**System information**
Mac M2 Pro
* Operating system:
* GMT version (`gmt --version`):
6.4.0

**Comment by joa-quim** [comment]
Sorry, what was it that gave
```
echo 90 89.6062907675 | mapproject -Js0/90/1:100 -C -R0/360/55/90
16762.4577062 -1.02640450879e-12
```
because I get
```
 echo 90 89.6062907675 | mapproject -Js0/90/1:100 -C -R0/360/55/90
43957.5269573   -2.69162223433e-12
```

**Comment by GregNeumann** [comment]
Your ELLIPSOID is probably WGS-84, I was working with a Mercury version,
but the problem is not with the ellipsoid. Going to the original
terrestrial gmt.conf, with WGS-84, I get your answer exactly, but it is the
inverse that is messing up. IE

echo 90 89 | mapproject -Js0/90/1:100 -C -R0/360/55/90

111652.02164	-6.83671454601e-12


but
echo 111652.02164 -6.83671454601e-12| mapproject -Js0/90/1:100 -C
-R0/360/55/90 -I

90 89.6062911089

I would expect the inverse to return 90 89 for lon/lat.

**Comment by GregNeumann** [comment]
PS I think you will find this lack of inverse behavior with the other polar
projections. Something to  do with the -I option.
Similarly, I find the behavior of mapproject -I in centered Mercator
coordinates also weird: (again terrestrial WGS-84)

***@***.*** ~ % echo 180 55 |mapproject -Jm1c -R0/360/-80/80 -C

0 65.8181030369


and the inverse should recover 180, 55 latitude but


***@***.*** ~ % echo 0 65.8181030369 |mapproject -Jm1c -R0/360/-80/80 -C
-I

180 0.000234345997728

**Comment by joa-quim** [comment]
> Your ELLIPSOID is probably WGS-84,

Sorry, you are right. I was doing one-liners and forgot to copy the other settings (ellipsoid, etc...)

**Comment by GregNeumann** [comment]
I hope it is a simple bug in "-I" processing and you can squash it easily.
Thanks for getting on the case. Greg

**Comment by joa-quim** [comment]
It works fine if we don't use **-C**

**Comment by joa-quim** [comment]
I submitted a PR that seems to fix this (#7239) but needs tests that it doesn't break other things.

**Comment by GregNeumann** [comment]
Correct, the bug appears only in centered projection inverses.
This was not the case in GMT 4 or GMT 5, where centered inverse projections
were useful to scope out sizes of projected objects near the pole.
